src/components/bookList/BookList.tsx

```diff
@@ -1,52 +1,69 @@
 import { useEffect, useState } from 'react'
 
+import { AxiosResponse } from 'axios'
 import InfiniteScroll from 'react-infinite-scroll-component'
 
 import { Book } from '../../model/Book'
-import BookRequest from '../../model/BookRequest'
 import BookService from '../../services/BookService'
 import Card from '../card/Card'
 import './BookList.css'
+import Search from '../search/Search'
+import Where from '../../model/Where'
+import BookResponse from '../../model/BookResponse'
 
-const numberOfCard = 24
+interface BooksRequest {
+  PageNumber: number,
+  PageLength: number,
+  Where: Where[],
+}
+const initialItem = 6
 const BookList = () => {
-  const [ book, setBook ] = useState<Book[]>([])
+  const [ books, setBooks ] = useState<Book[]>([])
   const [ hasMore, setHasMore ] = useState(true)
-  const [ pageNubmer, setPageNumber ] = useState(1)
-
-  useEffect(() => {
-    getDataBooks()
-  }, [ pageNubmer ])
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ searchValue, setSearchValue ] = useState('')
+  const [ currentSearch, setCurrentSearch ] = useState<string>(searchValue)
 
   const nexPage = () => {
-    setPageNumber( pageNubmer + 1)
+    setPageNumber(pageNumber => pageNumber + 1)
   }
 
-  const getDataBooks = async () => {
-    const bookRequest : BookRequest = {
-      PageNumber: pageNubmer,
-      PageLength: numberOfCard
-    }
-    try{
-      const response = await BookService.getAllBooks(bookRequest)
-      setHasMore(pageNubmer * numberOfCard <= response.data.TotalCount)
-      setBook([ ...book,...response.data.Items ])
-    } catch(error) {
-      console.error(error)
+  const dataBooks = () => {
+    const bookRequest : BooksRequest = {
+      PageNumber: pageNumber,
+      PageLength: initialItem,
+      Where: [ { Field: 'Title',Value: searchValue, Operation: 2 } ]
     }
+    BookService.getAllBooks(bookRequest)
+      .then((response:AxiosResponse<BookResponse>) => {
+        const totalElements = response.data.TotalCount
+        setHasMore(pageNumber * initialItem <= totalElements)
+        setBooks((books) => [ ...books,...response.data.Items ])
+      })
+      .catch((e) => console.error (e))
```

Comment:
Handle errors differently, console.error is not the best way, you dont give any feedback to user that something went wrong...

Reply:
Yes i know, I fix this problem when i finished return and rent features.

Reply:
But why dont you implement better solution for this new code and then later you have less places to change? You can change it later but isn't it more work than changing it as you work on these files?